packages/service-backend/src/ai/ai.service.ts

```diff
@@ -64,15 +72,132 @@ export class AiService {
+        parts: [{
+          text: `You are an AI assistant designed to create multiple-choice questions from PDF files. You will receive a PDF file and the desired number of questions as input.
+
+    **Your Task**
+
+    1. **Validate Input:**
+       - Ensure the ${AllowedMimeType[file.mimeType]} file is readable and contains sufficient information for question generation.
+       - Check that the requested number of questions is a positive integer. If not, return an error message: { "error": "Invalid number of questions. Please provide a positive integer." }.
+
+    2. **Extract Information:**
+       - Analyze the ${AllowedMimeType[file.mimeType]} to identify key concepts, facts, and relationships.
+
+    3. **Formulate Questions:**
+       - Craft clear, concise, and relevant multiple-choice questions that test the user's understanding of the material.
+       - Provide four answer choices for each question, with only one correct answer.
+
+    4. **Handle Errors:**
+       - If the ${AllowedMimeType[file.mimeType]} is unreadable, lacks sufficient content, or there's an issue generating the specified number of questions, return an error message: { "error": "Your specific error message here" }.
+
+    5. **Return Output:**
+       - If successful, return the questions as a JSON in the following format (the JSON should be serialized without spaces or newlines):
+
+       \`\`\`json
+        [
+          {
+            "question": "...",
+            "answers": [
+              { "answer": "...", "correct": true },
+              { "answer": "...", "correct": false },
+              { "answer": "...", "correct": false },
+              { "answer": "...", "correct": false }
+            ]
+          },
+          ... (more questions)
+        ]
+       \`\`\`
+         `
+        }]
       }
     };
-    const response = await AiService.generativeModel.generateContent(req);
+
+
+    let response;
+    try {
+      response = await AiService.generativeModel.generateContent(req);
+      console.log(JSON.stringify(response, null, 2));
+    } catch (e) {
+      Logger.error(e)
+      throw new ConflictException('Failed to generate questions');
+    }
+
     const data = response.response.candidates[0].content.parts[0].text;
-    const json = data.match(/```json(.*?)```/s);
-    if (!json) {
-      throw new Error('Failed to generate text');
+
+    if (data.includes('error')) {
```

Comment:
what if the questions are about errors?
Or does it mean if there is a key 'error'

Reply:
The ai will answer a json with an error as key.  it's in the system instructions and I also catch the error if the ai failed internally

Reply:
Perfect!